mysql-test/suite/galera/t/pxc_async_conn_failover.test

```diff
@@ -0,0 +1,369 @@
+# ==== Purpose ====
+# This tests ensures that PXC builds allows asynchronous connection UDFs while
+# operating in cluster mode.
+#
+# Ref: WL#12649: Automatic reconnection for Async replication between InnoDB
+#      Clusters
+#
+# It tests that asynchronous replication connection is re-established to new
+# sender in case the current asynchronous replication connection fails.
+#
+# ==== Implementation ====
+# 1. Create a chain replication between server 1, 2, 3 and 4 (all servers with
+#    wsrep loaded)
+# 2. Insert data on server1 and verify that its synced to the end of
+#    chain i.e. server4
+# 3. Verify Asynchronous Replication Connection Failover is initially
+#    disabled. Also verify Asynchronous Replication is initially
+#    connected to server 3.
+# 4. Enable Asynchronous Replication Connection Failover and set retry
+#    count and retry time to 2 and verify its enabled in
+#    performance_schema.replication_connection_configuration.
+# 5. Stop server 3.
+# 6. Verify that asynchronous replication fails with error
+#    ER_RPL_ASYNC_RECONNECT_FAIL_NO_SOURCE as there is no source added to
+#    table performance_schema.replication_asynchronous_connection_failover to
+#    connect.
+# 7. Verify that asynchronous replication tries connecting to server 3
+#    two (retry_count=2) times before it fails.
+# 8. Add three asynchronous connection failover servers and
+#    verify that they are inserted successfully in
+#    performance_schema.replication_asynchronous_connection_failover.
+# 9. Restart replica IO so asynchronous replication connection failover
+#    can execute and connect to new sender/source.
+# 10. Verify that asynchronous replication channel gets connected to
+#     server 2 and also verify that asynchronous replication tries
+#     connecting to server 3 two (retry_count=2) times.
+# 11. Insert more data on server1 and verify that its synced to the end
+#     of chain i.e. server 4.
+# 12. Stop server 2.
+# 13. Verify that asynchronous replication channel gets connected to
+#     server 1 and also verify that asynchronous replication tries
+#     connecting to server 2 two (retry_count=2) times.
+# 14. Insert more data on server1 and verify that its synced to the
+#     server 4.
+# 15. Stop server 1.
+# 16. Even after all source are stopped the replica IO thread keeps
+#     on re-trying to connect with all logged source in
+#     performance_schema.replication_asynchronous_connection_failover
+#     table. It tries connecting to server 3, 2 and again server 3.
+# 17. Stop replica IO
+# 18. Clean up
+#
+--source include/big_test.inc
+--source include/have_wsrep_provider.inc
+--source include/have_debug.inc
+--source include/have_debug_sync.inc
+
+--echo #
+--echo # 1. Create a chain replication between server 1, 2, 3 and 4.
+--echo #.
+--let $rpl_skip_reset_master_and_slave= 1
+--let $rpl_server_count= 4
```

Comment:
The whole file: Inconsistent spaces around `=`

Reply:
Done. Used `= ` for --let mtr assignments and `  =  ` for SQL statements.